MACS Compatibility

fkrisi11 opened the pull request:
New settings group: Animator Tweaks
- MACS Compatible Layer Icons (New Feature)
- Show Empty Layer Icon (Now toggleable)
- Show Layer WD State (Now toggleable)
- Show Mixed WD Layer Icon (Now toggleable)
- Animator Info (Now toggleable)
- Empty Layer Icon Color (New Feature)
- WD Layer Icon Color (New Feature)

Animator Graph Labels Settings Group
- Hold ALT to see all labels at any time (Now toggleable)

konsti219 commented:
I love seeing more features being added to tools I use every day, but I noticed three things:

1. I think compatibility options should be more towards the bottom of lists. Try just reordering the items which you added.
2. You added a setting for toggling items in the layer list, but the setting `LayerListShow(Mixed)WD` already existed, it was just never implemented. Your code could be adjusted to use the old setting, or the old one can just be removed. Also, your new properties in the preferences could be placed below the old setting, instead of putting it at the top, new code it typically added at the bottom.
3. Some `if`s could be combined (`if (a && b) { ... }` instead of `if (a) { if (b) { ... } }`).

fkrisi11 commented:
There we go. Now things should be better.
Thanks for the suggestions, @konsti219 !

fkrisi11 commented:
Renamed the MACS compatibility variable, and gave it 2 new features.

It now skips the auto scrolling patch in RATS if MACS compatibility is enabled, and you add:
- A new layer, since it breaks the MACS version of it
- A new parameter, as it made sense to also do that

These patches can be toggled on and off inside of the MACS settings, which is where the user should toggle them, if they don't want them.
Added these changes here, because this is meant to be a PR to make RATS more compatible with MACS, so it made sense.

rrazgriz commented:
Thanks for submitting. Looks good to me, but I don't own MACS. Let me know if further changes are needed and I can try to merge those in too.
